lib/buffy/throttle_and_timed.ex

```diff
@@ -0,0 +1,372 @@
+# credo:disable-for-this-file Credo.Check.Refactor.LongQuoteBlocks
+defmodule Buffy.ThrottleAndTimed do
```

Comment:
I feel like we could come up with a slightly better name for this to describe the use case but I am struggling to land on one. Maybe `ThrottleWithTimeout`, or `ThrottleWithLoop`? I think we can keep it as is for now and iterate on it later though unless we can come up with a better option without spending too much time on it.

Reply:
`ThrottleWithLoopInterval`?